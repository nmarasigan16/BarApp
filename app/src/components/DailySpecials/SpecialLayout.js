import React from 'react';
import PropTypes from 'prop-types';
import {
    FlatList,
} from 'react-native';
import SpecialRowComponent from "./SpecialRowComponent";

const SpecialLayout = ({specials}) => {
    const tempSpecials = [
        {
            barName: 'Legends',
            description: '$3 for any draft',
            barId: 5
        },
        {
            barName: 'Brothers',
            description: '$1 wells',
            barId: 4
        }
    ];

    specials = specials || tempSpecials;

    return (
        <FlatList
            data={specials}
            keyExtractor={(item, index) => item.barId}
            renderItem={({item}) => <SpecialRowComponent special={item}/>}
            />
    )

};

SpecialLayout.propTypes = {
    specials: PropTypes.array
};

export default SpecialLayout;
