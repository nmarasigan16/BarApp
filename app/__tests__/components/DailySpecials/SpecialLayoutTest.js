import 'react-native';
import React from 'react';
import SpecialLayout from '../../../src/components/DailySpecials/SpecialLayout';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const tempSpecials = [
    {
        barName: 'Legends',
        description: '$3 for any draft',
        barId: 5
    },
    {
        barName: 'Brothers',
        description: '$1 wells',
        barId: 4
    }
];

it('renders correctly with specials', () => {
    const tree = renderer.create(
        <SpecialLayout specials={tempSpecials}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

it('render correctly without specials', () => {
    const tree = renderer.create(
        <SpecialLayout/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
