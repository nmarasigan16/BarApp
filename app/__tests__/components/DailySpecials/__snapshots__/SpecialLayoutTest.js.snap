// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly without specials 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "barId": 5,
        "barName": "Legends",
        "description": "$3 for any draft",
      },
      Object {
        "barId": 4,
        "barName": "Brothers",
        "description": "$1 wells",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onViewableItemsChanged={undefined}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginLeft": 25,
            "marginRight": 25,
            "marginTop": 18.75,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "marginBottom": 18.75,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "Courier New",
                  },
                  Object {
                    "color": "silver",
                    "fontSize": 50,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Legends
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "Courier New",
                  },
                  Object {
                    "color": "darkgray",
                    "fontSize": 37.5,
                  },
                ]
              }
            >
              $3 for any draft
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#D3D3D3",
                "height": 1,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginLeft": 25,
            "marginRight": 25,
            "marginTop": 18.75,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "marginBottom": 18.75,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "Courier New",
                  },
                  Object {
                    "color": "silver",
                    "fontSize": 50,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Brothers
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "Courier New",
                  },
                  Object {
                    "color": "darkgray",
                    "fontSize": 37.5,
                  },
                ]
              }
            >
              $1 wells
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#D3D3D3",
                "height": 1,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`renders correctly with specials 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "barId": 5,
        "barName": "Legends",
        "description": "$3 for any draft",
      },
      Object {
        "barId": 4,
        "barName": "Brothers",
        "description": "$1 wells",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onViewableItemsChanged={undefined}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginLeft": 25,
            "marginRight": 25,
            "marginTop": 18.75,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "marginBottom": 18.75,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "Courier New",
                  },
                  Object {
                    "color": "silver",
                    "fontSize": 50,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Legends
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "Courier New",
                  },
                  Object {
                    "color": "darkgray",
                    "fontSize": 37.5,
                  },
                ]
              }
            >
              $3 for any draft
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#D3D3D3",
                "height": 1,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "marginLeft": 25,
            "marginRight": 25,
            "marginTop": 18.75,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
                "marginBottom": 18.75,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "Courier New",
                  },
                  Object {
                    "color": "silver",
                    "fontSize": 50,
                    "fontWeight": "bold",
                  },
                ]
              }
            >
              Brothers
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "Courier New",
                  },
                  Object {
                    "color": "darkgray",
                    "fontSize": 37.5,
                  },
                ]
              }
            >
              $1 wells
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#D3D3D3",
                "height": 1,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
